### PR TITLE
Adds a print report functions to Handheld Chemistry Equipment

### PR DESCRIPTION
**Pull request by Fruerlund**
Mass Spectrometers and Reagent Scanners will now print reports if users
choose to via an action button

As inspired by @TullyBurnalot autopsy PR

example
example

:cl: Fruerlund
add: adds an option to mass spectrometers and reagent scanners to print a report with the scan data
edit: renames the item_action from "Print Forensic Report" to "Print Report" so it can be used for the handheld chemistry equipment too
:cl:

### CODE REVIEW

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -434,6 +436,19 @@ REAGENT SCANNER
 	details = 1
 	origin_tech = "magnets=4;biotech=2"
 
+/obj/item/device/mass_spectrometer/verb/print_report()
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/scanners.dm]: Please stop making object verbs.

Comment (Fruerlund) [inline code/game/objects/items/devices/scanners.dm]: What do you want me to do instead?

Comment (Fruerlund) [inline code/game/objects/items/devices/scanners.dm]: Will fix.

Comment (Fox-McCloud) [inline code/game/objects/items/devices/scanners.dm]: Thanks!

**Comment by Fruerlund** [comment]
@Fox-McCloud fixed the object verb, I am not sure if I should stick with this method or the one used in the Autopsy PR.

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -478,11 +491,18 @@ REAGENT SCANNER
 					recent_fail = 1
 		if(dat)
 			to_chat(user, "<span class='notice'>Chemicals found: [dat]</span>")
+			if(alert("Do you want to print a log file?","Scan Results","Yes", "No") == "Yes")
```
Comment (Crazylemon64) [inline code/game/objects/items/devices/scanners.dm]: What happens if someone drops the scanner in a disposals bin and waits for it to be halfway across the station before saying "yes"?

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -424,8 +424,22 @@ REAGENT SCANNER
 					return
 				else
 					recent_fail = 1
-		to_chat(user, "[dat]")
-		reagents.clear_reagents()
+		if(dat)
+			to_chat(user, "Trace chemicals detected in blood sample: [dat]")
+			if(alert("Do you want to print a log file?","Scan Results","Yes", "No") == "Yes")
```
Comment (Crazylemon64) [inline code/game/objects/items/devices/scanners.dm]: Likewise

**Comment by Fruerlund** [comment]
@Crazylemon64 

Whole thing now uses action buttons instead.

File: code/game/objects/items/devices/scanners.dm (hunk 2 above)
Comment (Fruerlund) [inline code/game/objects/items/devices/scanners.dm]: Should be fixed by the action button

**Pull request merged**